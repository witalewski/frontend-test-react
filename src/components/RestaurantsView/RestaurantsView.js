import React, { Component, Fragment } from 'react';
import { shape, string } from 'prop-types';
import { inject, observer, PropTypes } from 'mobx-react';

class RestaurantsView extends Component {
  static propTypes = {
    restaurants: PropTypes.arrayOrObservableArrayOf(
      shape({
        id: string.isRequired,
        name: string.isRequired,
      })
    ).isRequired,
  };
  render() {
    return (
      <Fragment>
        <div className="hero">
          <h1>Restaurants</h1>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
            eiusmod tempor incididunt ut labore et dolore magna aliqua.
          </p>
        </div>
        <div className="restaurants-list">
          <ul>
            {this.props.restaurants.map(({ id, name }) => (
              <li key={id}>{name}</li>
            ))}
          </ul>
        </div>
      </Fragment>
    );
  }
}

export { RestaurantsView };
export default inject(({ appState }) => ({
  restaurants: appState.restaurants,
  setRestaurants: appState.setRestaurants,
}))(observer(RestaurantsView));
