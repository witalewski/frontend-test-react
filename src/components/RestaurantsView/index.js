export { default as RestaurantsView } from './RestaurantsView';