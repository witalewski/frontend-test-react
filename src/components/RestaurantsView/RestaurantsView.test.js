import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { RestaurantsView } from './RestaurantsView';

describe('RestaurantsView', () => {
  it('matches snapshot', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <RestaurantsView />
    );
    const result = renderer.getRenderOutput();

    expect(result).toMatchSnapshot();
  });
});
