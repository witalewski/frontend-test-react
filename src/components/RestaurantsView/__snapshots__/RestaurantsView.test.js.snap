// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantsView matches snapshot 1`] = `
<div
  className="restaurants-view"
>
  <div
    className="hero"
  >
    <h1>
      Restaurants
    </h1>
  </div>
  <inject-Filters />
  <inject-RestaurantsList />
</div>
`;
